### ISSUE TITLE
Inconsistent behavior when injecting components

### ISSUE THREAD

**Issue by tulioag** [vaadin/spring#344]
Suppose I have a template with a div injected

```
public class ExampleTemplate extends PolymerTemplate<ExampleTemplate.ExampleModel> {
    @Id("content") Div content;
```
Normally, a `com.vaadin.flow.component.html.Div` will be injected.

If, however, on the same package or on a subpackage I have a component like this

```
@SpringComponent
@UIScope
class ComplexComponent extends Div { }
```
It will now be injected everywhere that expects a normal Div, like the `ExampleTemplate` above.

Finally, if I create a second component, like so
```
@SpringComponent
@UIScope
class ComplexComponent2 extends Div {

}
```
Then neither `ComplexComponent` nor `ComplexComponent2` are injected in `ExampleTemplate,` but a normal `com.vaadin.flow.component.html.Div`. This behavior is confusing.

**Expected behavior:** Always inject a normal `com.vaadin.flow.component.html.Div`, unless the field requires a subclass.

Tested with platform 10.0.5.

**Comment by Artur-** [comment]
Handled by https://github.com/vaadin/spring/blob/3.0/vaadin-spring/src/main/java/com/vaadin/spring/internal/SpringBeanUtil.java#L38 in version 3.0

**Comment by denis-anisimov** [comment]
Fixed in `3.0` (see Artur's comment above).
